src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -2822,6 +2829,67 @@ void EmitAtomicSingleCharLoop(RegexNode node)
                     }
                     Stloc(iterationLocal);
                 }
+                else if (node.Type == RegexNode.Setloopatomic &&
+                    maxIterations == int.MaxValue &&
+                    (numSetChars = RegexCharClass.GetSetChars(node.Str!, setChars)) > 1 &&
+                    RegexCharClass.IsNegated(node.Str!))
```

Comment:
Don't we need to check `&& !IsCaseInsensitive(node)` as well?

Reply:
Good catch. Yes. I convinced myself we didn't for some reason, and I have no idea now why :-)